src/bgw/job_stat_history.c

```diff
@@ -192,34 +202,51 @@ bgw_job_stat_history_scan_id(int64 bgw_job_history_id, tuple_found_func tuple_fo
 }
 
 static ScanTupleResult
-bgw_job_stat_history_tuple_mark_end(TupleInfo *ti, void *const data)
+bgw_job_stat_history_tuple_update(TupleInfo *ti, void *const data)
 {
 	bool should_free;
 	HeapTuple tuple = ts_scanner_fetch_heap_tuple(ti, false, &should_free);
 	BgwJobStatHistoryContext *context = (BgwJobStatHistoryContext *) data;
+	Jsonb *job_history_data = NULL;
 
 	Datum values[Natts_bgw_job_stat_history] = { 0 };
 	bool nulls[Natts_bgw_job_stat_history] = { 0 };
 	bool doReplace[Natts_bgw_job_stat_history] = { 0 };
 
-	values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_pid)] = Int32GetDatum(MyProcPid);
-	doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_pid)] = true;
-
-	values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_execution_finish)] =
-		TimestampTzGetDatum(ts_timer_get_current_timestamp());
-	doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_execution_finish)] = true;
-
-	values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_succeeded)] =
-		BoolGetDatum((context->result == JOB_SUCCESS));
-	doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_succeeded)] = true;
-
-	Jsonb *job_history_data = ts_bgw_job_stat_history_build_data_info(context);
-
-	if (job_history_data != NULL)
+	switch (context->update_type)
 	{
-		values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_data)] =
-			JsonbPGetDatum(job_history_data);
-		doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_data)] = true;
+		case JOB_STAT_HISTORY_UPDATE_PID:
+		{
+			values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_pid)] =
+				Int32GetDatum(MyProcPid);
+			doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_pid)] = true;
+			break;
+		}
+
+		case JOB_STAT_HISTORY_UPDATE_END:
+		{
+			values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_execution_finish)] =
+				TimestampTzGetDatum(ts_timer_get_current_timestamp());
+			doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_execution_finish)] = true;
+
+			values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_succeeded)] =
+				BoolGetDatum((context->result == JOB_SUCCESS));
+			doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_succeeded)] = true;
+
+			job_history_data = ts_bgw_job_stat_history_build_data_info(context);
+
+			if (job_history_data != NULL)
+			{
+				values[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_data)] =
+					JsonbPGetDatum(job_history_data);
+				doReplace[AttrNumberGetAttrOffset(Anum_bgw_job_stat_history_data)] = true;
+			}
+			break;
+		}
+
+		case JOB_STAT_HISTORY_UPDATE_START:
```

Comment:
This switch is missing a default case. It is likely to trigger warnings in compilers and tools
```suggestion
        default:
		case JOB_STAT_HISTORY_UPDATE_START:
```

Reply:
Fixed!